datafusion/functions/src/core/arrow_cast.rs

```diff
@@ -147,10 +147,10 @@ static DOCUMENTATION: OnceLock<Documentation> = OnceLock::new();
 
 fn get_arrow_cast_doc() -> &'static Documentation {
     DOCUMENTATION.get_or_init(|| {
-        Documentation::builder()
-            .with_doc_section(DOC_SECTION_OTHER)
-            .with_description("Casts a value to a specific Arrow data type.")
-            .with_syntax_example("arrow_cast(expression, datatype)")
+        Documentation::builder(
+            DOC_SECTION_OTHER,
```

Comment:
it is somewhat unfortunate that we lose the builder syntax but I think it is worth it

Reply:
thanks @alamb. imho, it is regular builder pattern for the object construction which has both mandatory fields(in constructor) and optional fields(set by builder).